### PR TITLE
Add tests for `nil?`

### PR DESCRIPTION
**Pull request by rob-3**

### CODE REVIEW

File: test/clojure/core_test/nil_questionmark.cljc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    0 false
+    "" false
+    "nil" false
+    :nil false
+    ##Inf false
+    ##NaN false
+    0.0 false
+    {} false
+    '() false
+    [] false))
```
Comment (jeaye) [inline test/clojure/core_test/nil_questionmark.cljc:16]: It'd be good to add an infinite sequence test, á la: https://github.com/jank-lang/clojure.core-test/blob/main/test/clojure/core_test/any.cljc#L13

Comment (rob-3) [reply test/clojure/core_test/nil_questionmark.cljc]: Done. I can add it in a separate `deftest` if you prefer. I also added a test for `false` because I realized I missed it the first time around.

Comment (jeaye) [reply test/clojure/core_test/nil_questionmark.cljc]: Nice! I think it's good to be explicit about it in a separate test. `common` is meant for normal cases, while other edge cases have their own tests.

Comment (rob-3) [reply test/clojure/core_test/nil_questionmark.cljc:17]: Ok - changed!

Comment (jeaye) [reply test/clojure/core_test/nil_questionmark.cljc:17]: :muscle: Thanks for helping!
